### PR TITLE
Add `logcosh`

### PR DESCRIPTION
**Pull request by devmotion**
This PRs adds an implementation of `log(cosh(x))`.

Fixes https://github.com/JuliaStats/LogExpFunctions.jl/issues/31.

cc @cossio

### CODE REVIEW

**Comment by cossio** [comment]
See my comment https://github.com/JuliaStats/LogExpFunctions.jl/issues/31#issuecomment-993395336. 

But anyway I think this could be merged? It's probably accurate enough for practical purposes.

**Comment by cossio** [comment]
Merge_

File: src/basicfuns.jl [hunk 1]
```diff
@@ -94,6 +94,16 @@ logit(x::Real) = log(x / (one(x) - x))
 """
 $(SIGNATURES)
 
+Return `log(cosh(x))`.
```
Comment (nalimilan) [inline src/basicfuns.jl:97]: Mention what's the difference with just calling `log(cosh(x))` (i.e. precision, `abs`, etc.)?

Comment (devmotion) [reply src/basicfuns.jl:97]:
```suggestion
Return `log(cosh(x))`, carefully evaluated without intermediate calculation of `cosh(x)`.

The implementation ensures `logcosh(-x) = logcosh(x)`.
```

Comment (devmotion) [reply src/basicfuns.jl]: I extended the docstring a bit :slightly_smiling_face:

**Review by nalimilan — APPROVED** [review]

**Pull request merged**